### ISSUE TITLE
[Bug]: Accessibility issue - in File menu when moving between files using screen readers (JAWS, Narrator, NVDA)

### ISSUE THREAD

**Issue by citygabor** [nextcloud/server#56611]
### ⚠️ This issue respects the following points: ⚠️

- [x] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [x] This issue is **not** already reported on [Github](https://github.com/nextcloud/server/issues?q=is%3Aopen+is%3Aissue+label%3Abug) OR [Nextcloud Community Forum](https://help.nextcloud.com/) _(I've searched it)_.
- [x] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [x] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

Hello NextCloud Developers and Community,

I have a vision loss and working with screen readers (JAWS, Narrator, NVDA).

When I open the files menu  I could navigate between the directories without any issues at the "Name" column, but when I moving my keyboard to the files section my screen readers (JAWS, Narrator, NVDA) only say "open/review without the file name. I have tested with the default files only.

Please check it by own with any of the listed screen readers.

NextCloud version: 32.x
Please let me know if you need any additional information from my side.

Thanks,
Regards,
Gabor

### Steps to reproduce

1. Open the File menu.
2. Navigate to the Directory and the file table menu with keyboard shortcut with "T" when my screen reader is on.
3. moving with the table keyboard combination. Ctrl+Alt+arrow keys to the "Name" column
4. When the Screen reader is  on a directory there is no issue. It can read the directory name without any issues.
5. But when the screen reader is on any of the default files it only says "open/review" without the name of the file.

### Expected behavior

The screen readers have to read the name of files as well same as the name of the directories.

### Nextcloud Server version

30

### Operating system

None

### PHP engine version

None

### Web server

None

### Database engine version

None

### Is this bug present after an update or on a fresh install?

None

### Are you using the Nextcloud Server Encryption module?

None

### What user-backends are you using?

- [ ] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### Configuration report

```json
N/A
```

### List of activated Apps

```shell
- Files
- Photos
- Activities
- Control panel
```

### Nextcloud Signing status

```shell
N/A
```

### Nextcloud Logs

```json
N/A
```

### Additional info

N/A

**Comment by susnux** [comment]
Do I understand the problem correctly?
- You open the files app (url: /apps/files)
- You navigate into the files list (table)
- You are in the "name" column
- You go to the next row
- If the row is a file then the filename is not announced but only "open"

**Comment by susnux** [comment]
I was able to reproduce this with Orca.
I am not sure what the best is as I am not a screenreader user, so @citygabor what do you think is the best solution here?
1. Put the instructions ("open" or "view") into the header and not duplicate this in every row. So rows just read as "file: xy" or "folder: xy"
2. Just fix the files entries to also have the name (just like folders) and keep everything as is

**Comment by citygabor** [comment]
Hello @susnux ,

Thanks so much for getting back to me,

Yes, there is a Linux screen reader what I forgot to mention, which is Orca. :-)

I am glad you could reproduce the issue.

Both senario would be fine for me, but I think number 2. has less job to fix the file name in the table view instead of option 1. where you need to hide the open/view announcement for screen readers. AS the other cells has similar context I would like to vote option 2. As we  can hear the action on the file or directory wen the focus is there.


One more thing I could read a "Keyboard navigation help" words inside the Photos or Files menu. This is just a html element and cannot be clicked, am I  right?

Thanks for your support,

**Comment by susnux** [comment]
> One more thing I could read a "Keyboard navigation help" words inside the Photos or Files menu. This is just a html element and cannot be clicked, am I right?

That should have the "skip to app navigation" button auto-focused (so you can directly skip either to app navigation or main content).

**Comment by javido** [comment]
Hello all, I can confirm this also with Orca screen reader.
The issue has to do with the aria-label attribute on the "open" button associated with the file name. 
The same behaviour with folder should be applied to files in the aria-label. Currently the aria-label is "open" and overrides the filename information that is in the button text. 
So, I would put the file name information also in the aria-label aria-label="Open filename" 

Many thanks and regards
Javier

**Comment by citygabor** [comment]
Hello @susnux ,

Thanks for your effort on this reported problem this is working now as expected.

at least with the NexCloud 34.x version.

Thanks again,
Regards,
Gabor
